Client/WarFare/UIUpgradeSelect.cpp

```diff
@@ -42,19 +42,13 @@ bool CUIUpgradeSelect::ReceiveMessage(
 	{
 		if (pSender == m_pBtn_Upgrade_1)
 		{
-#if 1
-			CGameProcedure::MessageBoxPost(
-				"CUIItemUpgrade needs to be implemented.",
-				"Not implemented",
-				MB_OK);
-#else
+
 			CUIItemUpgrade* pUIItemUpgrade = CGameProcedure::s_pProcMain->m_pUIItemUpgrade;
 			if (pUIItemUpgrade != nullptr)
 			{
-				pUIItemUpgrade->SetVisible(true);
-				pUIItemUpgrade->SetNpcID(m_iNpcID);
+				pUIItemUpgrade->Open();
+				// pUIItemUpgrade->SetNpcID(m_iNpcID);
```

Comment:
This was a thing because of the packet structure. The NPC ID as associated here is passed in the upgrade packet.
Considering you end up placeholdering it out, I suggest restoring it and using it as was intended:
```cpp
	uint16_t sNpcID = 1;
	CAPISocket::MP_AddByte(byBuff, iOffset, sNpcID);
```